substrafl/algorithms/pytorch/torch_scaffold_algo.py

```diff
@@ -434,6 +434,8 @@ def train(
                 "self._model(data) is called) after the `self.optimizer.step()` call."
             )
 
+        self._reset_scaffold_parameters_update()
```

Comment:
Missing the reset cause it was a side effect of the "save" and "load" methods, that was reimplementing the object so reseting it by design.